### ISSUE TITLE
describe_nat_gateways is not a PageableResponse

### ISSUE THREAD

**Issue by ryanbrainard** [aws/aws-sdk-ruby#1408]
I am trying to page over the results of several different resource types with [Aws::PageableResponse](https://docs.aws.amazon.com/sdkforruby/api/Aws/PageableResponse.html). This is working fine for everything except for the`describe_nat_gateways` responses.

For example, I can do:

```ruby
ec2.describe_instances.each { |page| puts page.reservations.count }
15
```

or 

```ruby
ec2.describe_vpcs.each { |page| puts page.vpcs.count }
3
```

but cannot do:

```
ec2.describe_nat_gateways.each { |page| puts page.nat_gateways.count }
NoMethodError: undefined method `each' for #<Seahorse::Client::Response:0x00562ff79eb038>
```

This is happening because it is not a `Aws::PageableResponse`:

```ruby
ec2.describe_nat_gateways.is_a?(Aws::PageableResponse)
false
```

unlike the others:

```ruby
ec2.describe_instances.is_a?(Aws::PageableResponse)
true

ec2.describe_vpcs.is_a?(Aws::PageableResponse)
true
```

Look both at the docs and `apis/ec2/2016-11-15/paginators-1.json`, it appears that it should work. Why is only describe NAT gateways not working?

As a side note, but potentially related to this problem, the default stubbed response from `describe_nat_gateways` is containing the literal word `String` as the `next_token` unlike all the others that default to `nil`.

Here are the versions I am using from my `Gemfile.lock`:

```
    aws-sdk-core (2.7.0)
      aws-sigv4 (~> 1.0)
      jmespath (~> 1.0)
    aws-sdk-resources (2.7.0)
      aws-sdk-core (= 2.7.0)
    aws-sigv4 (1.0.0
```

**Comment by cjyclaire** [comment]
@ryanbrainard Thanks for reporting! Looks like [`ec paginators`](https://github.com/aws/aws-sdk-ruby/blob/master/aws-sdk-core/apis/ec2/2016-11-15/paginators-1.json) is missing `DescribeNatGateways` exactly, that's the cause. This is definitely something we need to fix. Yet introducing pagination behavior for existing APIs might be a breaking change to other customers. We will try to figure out a way to fix this from root.
Meanwhile, as quick workaround, you could try `response.to_a.each ..` or  `response.to_h.each ...` Yet in this way, other benefits of paginators will not be enjoyed.

**Comment by awood45** [comment]
We're going to add that paginator, hopefully to go out with the next release. I'll close this issue once that new paginator is in place.

**Comment by ryanbrainard** [comment]
Thanks for the quick response. Will the fix for the paginator also fix the "String" being returned for next_token in stubs or should I file a separate issue for that?

**Comment by awood45** [comment]
We can use a separate issue to discuss that.

**Comment by cjyclaire** [comment]
A [tagged](https://github.com/aws/aws-sdk-ruby/releases/tag/v2.7.8) release containing the fix is out, closing.
